### PR TITLE
Coreutils: Added wc Utility

### PR DESCRIPTION
**Pull request by circutrider21**
Added the wc utility to Coreutils. Not much else to say.

### CODE REVIEW

File: applications/coreutils/wc.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+  for(int c = 0; c < argc; c++) {
+    if(!checkArg(argv[c], params)) {
+      printFileInfo(params, all, wc(argv[c]));
+    }
+  }
+
+  return PROCESS_SUCCESS;
+}
```
Comment (sleepy-monax) [inline applications/coreutils/wc.cpp:136]: All of this can be handled by libutils/ArgParse.h

**Comment by Keyboard-Slayer** [comment]
Please read the [Contributing guide](https://github.com/skiftOS/skift/blob/master/contributing.md)

File: applications/coreutils/wc.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  __cleanup(stream_cleanup) Stream *stream = stream_open(path, OPEN_READ);
+  if (handle_has_error(stream)) {
+    process_exit(PROCESS_ERROR);
+  }
+
+  // Get number of bytes from file
+
+  FileState stat = {};
+  stream_stat(stream, &stat);
+  to_return.bytes = stat.size;
```
Comment (sleepy-monax) [inline applications/coreutils/wc.cpp:79]: This won't work if you pipe into wc

File: applications/coreutils/wc.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+  char buffer[stat.size];
```
Comment (sleepy-monax) [inline applications/coreutils/wc.cpp:81]: Unbounded vla = big no no

File: applications/coreutils/wc.cpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  size_t read;
+  read = stream_read(stream, &buffer, stat.size)
+  
+  to_return.lines = line_count(buffer);
+  to_return.words = word_count(buffer);
+  to_return.chars = strlen(buffer);
```
Comment (sleepy-monax) [inline applications/coreutils/wc.cpp:87]: Why are you iterating 3 times on the input, also file and stream can contain \0

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

File: makefile [hunk 5]
```diff
@@ -5,9 +5,6 @@ export PATH := $(shell toolchain/use-it.sh):$(PATH)
 export PATH := $(shell toolbox/use-it.sh):$(PATH)
 export LC_ALL=C
```
Comment (Keyboard-Slayer) [inline makefile:6]: You shouldn't edit that

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: makefile [hunk 6]
```diff
@@ -59,29 +56,31 @@ BUILD_DEFINES:= \
 	-D__BUILD_VERSION__=\""$(CONFIG_VERSION)"\"
 
 # --- Configs -------------------------------------------- #
-CC:=i686-pc-skift-gcc
+CC:=gcc
```
Comment (sleepy-monax) [inline makefile:59]: why

File: makefile [hunk 7]
```diff
@@ -59,29 +56,31 @@ BUILD_DEFINES:= \
...
 CFLAGS= \
 	-std=gnu11 \
+  -m32 \
```
Comment (sleepy-monax) [inline makefile:62]: you broke the x86_64 build by doing that

File: makefile [hunk 8]
```diff
@@ -59,29 +56,31 @@ BUILD_DEFINES:= \
...
 	-MD \
 	$(BUILD_OPTIMISATIONS) \
 	$(BUILD_WARNING) \
 	$(BUILD_INCLUDE) \
 	$(BUILD_DEFINES) \
 	$(BUILD_CONFIG)
 
-CXX:=i686-pc-skift-g++
+CXX:=g++
```
Comment (sleepy-monax) [inline makefile:70]: again

File: makefile [hunk 9]
```diff
@@ -59,29 +56,31 @@ BUILD_DEFINES:= \
...
 CXXFLAGS:= \
 	-std=c++20 \
+  -m32 \
```
Comment (sleepy-monax) [inline makefile:73]: this is painfull

File: makefile [hunk 10]
```diff
@@ -59,29 +56,31 @@ BUILD_DEFINES:= \
...
 	-MD \
 	$(BUILD_OPTIMISATIONS) \
 	$(BUILD_WARNING) \
 	$(BUILD_INCLUDE) \
 	$(BUILD_DEFINES)
 
-LD:=i686-pc-skift-ld
+LD:=ld
```
Comment (sleepy-monax) [inline makefile:80]: this is outside the stated scope of your pr

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

File: makefile (hunk 6 above)
Comment (circutrider21) [reply makefile:59]: I didn't mean for my changes to the makefile to be included into the pull request. Must of happened by accident. Oops.

**Comment by circutrider21** [comment]
I didn't mean for my changes to the makefile to be included into the oil request. Don't know how that happened.

File: applications/coreutils/wc.cpp (hunk 2 above)
Comment (circutrider21) [reply applications/coreutils/wc.cpp:79]: Do you have any other ideas on how I could get the size of the file in bytes then?

**Pull request closed without merging**